controllers/cluster_controller.go

```diff
@@ -153,23 +153,37 @@ func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager, log logr.Logger)
+// +kubebuilder:rbac:groups="",resources=events;secrets;secrets/status;configmaps;configmaps/status;namespaces;namespaces/status;services;services/status,verbs=get;list;watch;
+// +kubebuilder:rbac:groups="",resources=node,verbs=*
+// +kubebuilder:rbac:groups="",resources=node/status,verbs=patch
+// +kubebuilder:rbac:groups="",resources=serviceaccounts;endpoints;persistentvolumes,verbs=get;list;watch
+// +kubebuilder:rbac:groups="",namespace=eksa-system,resources=secrets,verbs=delete;
+// +kubebuilder:rbac:groups=anywhere.eks.amazonaws.com,resources=gitopsconfigs;clusters;snowmachineconfigs;snowdatacenterconfigs;snowippools;vspheredatacenterconfigs;vspheremachineconfigs;dockerdatacenterconfigs;tinkerbellmachineconfigs;tinkerbelldatacenterconfigs;cloudstackdatacenterconfigs;cloudstackmachineconfigs;nutanixdatacenterconfigs;nutanixmachineconfigs;bundles;awsiamconfigs,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=anywhere.eks.amazonaws.com,resources=clusters/status;snowmachineconfigs/status;snowippools/status;vspheredatacenterconfigs/status;vspheremachineconfigs/status;dockerdatacenterconfigs/status;tinkerbelldatacenterconfigs/status;tinkerbellmachineconfigs/status;cloudstackdatacenterconfigs/status;cloudstackmachineconfigs/status;bundles/status;awsiamconfigs/status,verbs=get;update;patch
 // +kubebuilder:rbac:groups=anywhere.eks.amazonaws.com,resources=oidcconfigs,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=anywhere.eks.amazonaws.com,resources=awsiamconfigs,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups=anywhere.eks.amazonaws.com,resources=clusters/status;snowmachineconfigs/status;snowippools/status;vspheredatacenterconfigs/status;vspheremachineconfigs/status;dockerdatacenterconfigs/status;cloudstackdatacenterconfigs/status;cloudstackmachineconfigs/status;nutanixdatacenterconfigs/status;nutanixmachineconfigs/status;bundles/status;awsiamconfigs/status;tinkerbelldatacenterconfigs/status;tinkerbellmachineconfigs/status,verbs=;get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups=anywhere.eks.amazonaws.com,resources=clusters/finalizers;snowmachineconfigs/finalizers;snowippools/finalizers;vspheredatacenterconfigs/finalizers;vspheremachineconfigs/finalizers;dockerdatacenterconfigs/finalizers;cloudstackdatacenterconfigs/finalizers;cloudstackmachineconfigs/finalizers;nutanixdatacenterconfigs/finalizers;nutanixmachineconfigs/finalizers;bundles/finalizers;awsiamconfigs/finalizers;tinkerbelldatacenterconfigs/finalizers;tinkerbellmachineconfigs/finalizers,verbs=get;list;watch;create;update;patch;delete
```

Comment:
The kubebuilder only reads annotation from legacy controller, so this is to mapping what we actually have from legacy controller.

Reply:
That's interesting. So we're essentially making sure that all the annotations we have here match what's in the legacy controller.

Will we change kubebuilder to read from the new controller instead of the legacy at some point? Is it better to do that now?